documentation updates (#1865)
openvinotoolkit/model_server: docs/build_from_source.md

```bash
make docker_build OV_USE_BINARY=0 OV_SOURCE_BRANCH=<commit or branch> OV_SOURCE_ORG=<fork org>
```

### `RUN_TESTS`

Enables or disabled unit tests execution as part of the docker image building.
- `0` Unit tests are skipped
- `1` Unit tests are executed (default)

```bash
make docker_build RUN_TESTS=0
```

Running the unit tests will make the building to last longer and it will consume a bit more RAM

### `CHECK_COVERAGE`

Enables or disabled calculating the unit tests coverage as part of the docker image building.
- `0` Checking the coverage is skipped
- `1` Checking the coverage is included

```bash
make docker_build RUN_TESTS=0
```

> Running the unit tests will increase build time and consume more RAM

Reviewer: Looks like a bit changed copy of the RUN_TEST section

> Running the unit tests will make the building to last longer and it will consume a bit more RAM

Reviewer 2: ```suggestion
Running the unit tests will increase build time and consume more RAM
```

### `JOBS`

> Number of compilation jobs. By default it is set to the number of CPU cores. On hosts with low RAM, this value can be reduced to avoid out of memory errors form the compilation.

Reviewer: ```suggestion
Number of compilation jobs. By default it is set to the number of CPU cores. On hosts with low RAM, this value can be reduced to avoid out of memory errors during the compilation.
```
